packages/strapi-design-system/src/Grid/GridItem.tsx

```diff
@@ -1,12 +1,20 @@
 import React from 'react';
 
-import PropTypes from 'prop-types';
 import styled from 'styled-components';
 
 import { useGrid } from './GridContext';
-import { Box } from '../Box';
-
-const GridItemWrapper = styled.div`
+import { Box, BoxProps } from '../Box';
+import type { ResponsiveValue } from '../helpers/handleResponsiveValues';
+
+export interface GridItemProps extends BoxProps<HTMLDivElement> {
+  col: number;
+  s: number;
+  xs: number;
+  gap: ResponsiveValue;
+  gridCols: number;
+}
+
+const GridItemWrapper = styled.div<GridItemProps>`
```

Comment:
If we don't refactor this to be a styled Box, then we need to pick the specific types to the ones we're using, I also don't see where `gap` or `gridCols` is used – is this even necessary?

Reply:
Which makes me think we're using context for absolutely no reason?